### PR TITLE
Simplify the RTC rpmsg driver implementation

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

- rtc/rpmsg: Replace session to client 
- rtc/rpmsg: Call up_rtc_set_lowerhalf inside rpmsg_rtc_[server_]initialize

## Impact
Refactor only

## Testing
Pass CI

### CODE REVIEW

File: drivers/timers/rpmsg_rtc.c [hunk 1]
```diff
@@ -653,31 +656,31 @@ static void rpmsg_rtc_server_ns_bind(FAR struct rpmsg_device *rdev,
...
 
   if (strcmp(name, RPMSG_RTC_EPT_NAME))
     {
       return;
     }
 
-  session = kmm_zalloc(sizeof(*session));
-  if (!session)
+  client = kmm_zalloc(sizeof(*client));
+  if (!client)
```
Comment (pkarashchenko) [inline drivers/timers/rpmsg_rtc.c:667]:
```suggestion
  if (client == NULL)
```

File: drivers/timers/rpmsg_rtc.c [hunk 2]
```diff
@@ -689,31 +692,26 @@ static void rpmsg_rtc_server_ns_bind(FAR struct rpmsg_device *rdev,
...
  ****************************************************************************/
 
 #ifndef CONFIG_RTC_RPMSG_SERVER
-FAR struct rtc_lowerhalf_s *rpmsg_rtc_initialize(int minor)
+FAR struct rtc_lowerhalf_s *rpmsg_rtc_initialize(void)
 {
   FAR struct rpmsg_rtc_lowerhalf_s *lower;
 
   lower = kmm_zalloc(sizeof(*lower));
   if (lower)
```
Comment (pkarashchenko) [inline drivers/timers/rpmsg_rtc.c:707]:
```suggestion
  if (lower != NULL)
```

File: drivers/timers/rpmsg_rtc.c [hunk 3]
```diff
@@ -590,38 +590,41 @@ static int rpmsg_rtc_server_ept_cb(FAR struct rpmsg_endpoint *ept,
...
+                                            struct rpmsg_rtc_client_s, ept);
         FAR struct rpmsg_rtc_alarm_set_s *msg = data;
         FAR struct rpmsg_rtc_server_s *server = priv;
         time_t time = msg->sec;
         struct lower_setalarm_s alarminfo =
         {
           .id = msg->id,
           .cb = rpmsg_rtc_server_alarm_cb,
-          .priv = session
+          .priv = client
```
Comment (pkarashchenko) [inline drivers/timers/rpmsg_rtc.c:627]: C89 incompatible

File: drivers/timers/rpmsg_rtc.c [hunk 4]
```diff
@@ -554,28 +554,28 @@ static int rpmsg_rtc_server_cancelperiodic
...
 
 #ifdef CONFIG_RTC_ALARM
 static void rpmsg_rtc_server_alarm_cb(FAR void *priv, int alarmid)
 {
-  FAR struct rpmsg_rtc_session_s *session = priv;
+  FAR struct rpmsg_rtc_client_s *client = priv;
   struct rpmsg_rtc_alarm_fire_s msg =
   {
     .header.command = RPMSG_RTC_ALARM_FIRE,
     .id = alarmid,
```
Comment (pkarashchenko) [inline drivers/timers/rpmsg_rtc.c:575]: C89 incompatible

File: drivers/timers/rpmsg_rtc.c (hunk 2 above)
Comment (xiaoxiang781216) [reply drivers/timers/rpmsg_rtc.c]: Done.

File: drivers/timers/rpmsg_rtc.c (hunk 4 above)
Comment (xiaoxiang781216) [reply drivers/timers/rpmsg_rtc.c]: Done.

File: drivers/timers/rpmsg_rtc.c (hunk 3 above)
Comment (xiaoxiang781216) [reply drivers/timers/rpmsg_rtc.c]: Done.

File: drivers/timers/rpmsg_rtc.c (hunk 1 above)
Comment (xiaoxiang781216) [reply drivers/timers/rpmsg_rtc.c]: Done.

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Pull request merged**